### PR TITLE
Task not allocated

### LINKED ISSUE #206: [PE-D] [W12-2] Duration for flexible tasks too long
When I add a task that lasts 254 hours long, start time isn't set as default, and it doesn't show up when I use the list command.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: thngyuxuan/ped#3

### PR DESCRIPTION
**Pull request by Kafcis**
Add exception to prevent tasks that take too long for a day

### CODE REVIEW

File: src/main/java/athena/task/Task.java [hunk 1]
```diff
@@ -264,6 +265,9 @@ public Time getTimeInfo() {
      */
     @Override
     public String toString() {
+        if (timeInfo.getStartTimeString()==""){
```
Comment (daniellimws) [inline src/main/java/athena/task/Task.java]: Use `.equals` to compare strings

File: src/main/java/athena/TaskList.java [hunk 2]
```diff
@@ -249,8 +254,10 @@ public Task deleteTask(int taskNumber) throws TaskNotFoundException {
     public void editTask(int taskNumber, String name, String startTime, String duration,
                          String deadline, String recurrence, Importance importance,
                          String notes)
-            throws TaskNotFoundException, ClashInTaskException,
-            TaskDuringSleepTimeException, InvalidRecurrenceException, InvalidDeadlineException {
+            throws
+            TaskNotFoundException, ClashInTaskException, TaskDuringSleepTimeException, DateHasPassedException,
+
```
Comment (wish2023) [inline src/main/java/athena/TaskList.java]: Empty line here

File: src/main/java/athena/task/Time.java [hunk 3]
```diff
@@ -84,9 +87,7 @@ private boolean isClashWithSleep() {
     }
 
     private boolean isNoClashWithSleep() {
-        // Needed only if we make sleep time customizable (Causing bugs) - Low priority
-        // LocalTime twoThreeFiveNine = LocalTime.of(23,59);
-        // System.out.println(endTime.compareTo(SLEEP_TIME) <= 0);
+        LocalTime twoThreeFiveNine = LocalTime.of(23, 59);
```
Comment (wish2023) [inline src/main/java/athena/task/Time.java]: You can remove this since it's not being used

File: src/main/java/athena/task/Time.java [hunk 4]
```diff
@@ -76,6 +74,11 @@ public Time(Boolean isFlexible, String startTime, String duration, String deadli
                 throw new TaskDuringSleepTimeException();
             }
         }
+        if (this.duration > 16) {
+            throw new TaskTooLongException(this.duration);
+        }
```
Comment (wish2023) [inline src/main/java/athena/task/Time.java]: Could you shift this into the isClashWithSleep method so it looks cleaner?

`if (duration > 16 && isFlexible) throw new TaskTooLongException(duration)`

**Review by wish2023 — CHANGES_REQUESTED** [review]
Looks okay! Small changes suggested

File: src/main/java/athena/TaskList.java [hunk 5]
```diff
@@ -74,7 +75,9 @@ public TaskList(ArrayList<Task> tasks) {
      */
     private Task createTask(int number, String name, String startTime, String duration, String deadline,
                             String recurrence, Importance importance, String notes, Boolean isFlexible)
-            throws TaskDuringSleepTimeException, InvalidRecurrenceException, InvalidDeadlineException {
+            throws TaskDuringSleepTimeException, InvalidRecurrenceException, InvalidDeadlineException,
+            TaskTooLongException {
+
```
Comment (wish2023) [inline src/main/java/athena/TaskList.java]: whitespace

File: src/main/java/athena/TaskList.java [hunk 6]
```diff
@@ -148,8 +151,9 @@ public Task addTask(int number, String name, String startTime, String duration,
     public Task addTask(String name, String startTime, String duration,
                         String deadline, String recurrence,
                         Importance importance, String notes, Boolean isFlexible)
-            throws ClashInTaskException, TaskDuringSleepTimeException,
-            InvalidTimeFormatException, InvalidRecurrenceException, InvalidDeadlineException {
+
```
Comment (wish2023) [inline src/main/java/athena/TaskList.java]: whitespace

File: src/main/java/athena/task/TimeData.java [hunk 7]
```diff
@@ -70,19 +76,24 @@ public Time(boolean isFlexible, LocalTime startTime, int duration, String deadli
...
-            throws TaskDuringSleepTimeException,
+    public TimeData(Boolean isFlexible, String startTime, String duration, String deadline, String recurrence)
+            throws TaskDuringSleepTimeException, DateTimeParseException, TaskTooLongException,
             InvalidRecurrenceException, InvalidDeadlineException {
-        setIsFlexible(isFlexible);
-        setDuration(duration);
-        setDeadline(deadline);
+        this.duration = Integer.parseInt(duration);
+        this.isFlexible = isFlexible;
+        this.deadline = deadline;
```
Comment (wish2023) [inline src/main/java/athena/task/TimeData.java]: Any reason why you changed this? I think it'd be better to call the `set` methods as it all follows a similar level of abstraction

**Review by wish2023 — CHANGES_REQUESTED** [review]
I almost forgot. It'd be great if you could handle the Ui error too when no start time is stated.
Screenshot 2020-11-09 at 12 20 30 AM

**Review by wish2023 — APPROVED** [review]

**Pull request merged**